core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/PhysicalPlan.scala

```diff
@@ -19,7 +19,7 @@ import org.jgrapht.util.SupplierUtil
 import scala.jdk.CollectionConverters.{IteratorHasAsScala, ListHasAsScala, SetHasAsScala}
 
 object PhysicalPlan {
-
+  def empty: PhysicalPlan = PhysicalPlan(operators = Set.empty, links = Set.empty)
```

Comment:
I see you are using empty physical plan in the case of compilation failure. I feel unnecessary to provide a valid empty physical plan, because it has a wrong semantic.  When failure happens, there should be no physical plan provided (None or undefined) to frontend. An empty physical plan does not mean it is invalid.

Reply:
removed